ext/dom/lexbor/lexbor/selectors-adapted/selectors.c

```diff
@@ -0,0 +1,1932 @@
+	return a->ns == b->ns || (a->ns != NULL && b->ns != NULL && xmlStrEqual(a->ns->href, b->ns->href));
+}
+
+static zend_always_inline bool lxb_selectors_adapted_cmp_local_name_id(const xmlNode *node, const lxb_selectors_adapted_id *id)
+{
+	uintptr_t ptr = (uintptr_t) node->name;
+	if (id->interned && (ptr & (ZEND_MM_ALIGNMENT - 1)) != 0) {
+		/* It cannot be a heap-allocated string because the pointer is not properly aligned for a heap allocation.
+		 * Therefore, it must be interned into the dictionary pool. */
+		return node->name == id->name;
+	}
+
+	return strcmp((const char *) node->name, (const char *) id->name) == 0;
+}
+
+static zend_always_inline const xmlAttr *lxb_selectors_adapted_attr(const xmlNode *node, const lxb_char_t *name)
+{
+	const xmlAttr *attr = xmlHasProp(node, (const xmlChar *) name);
+	if (attr != NULL && attr->ns != NULL) {
+		return NULL;
+	}
+	return attr;
+}
+
+static zend_always_inline bool lxb_selectors_adapted_has_attr(const xmlNode *node, const char *name)
+{
+	return lxb_selectors_adapted_attr(node, (const lxb_char_t *) name) != NULL;
+}
+
+static zend_always_inline const lxb_char_t *lxb_selectors_adapted_attr_value_raw(const xmlAttr *attr)
+{
+	ZEND_ASSERT(attr != NULL);
+
+	if (attr->children == NULL) {
+		return NULL;
+	}
+
+	return (const lxb_char_t *) attr->children->content;
+}
+
+static zend_always_inline lexbor_str_t lxb_selectors_adapted_attr_value_nullable(const xmlAttr *attr)
+{
+	lexbor_str_t ret;
+	ret.data = (lxb_char_t *) lxb_selectors_adapted_attr_value_raw(attr);
+	ret.length = (ret.data == NULL) ? 0 : strlen((const char *) ret.data);
+	return ret;
+}
+
+static zend_always_inline lexbor_str_t lxb_selectors_adapted_attr_value_empty(const xmlAttr *attr)
+{
+	lexbor_str_t ret;
+	ret.data = (lxb_char_t *) lxb_selectors_adapted_attr_value_raw(attr);
+	if (ret.data == NULL) {
+		ret.data = (lxb_char_t *) "";
+		ret.length = 0;
+	} else {
+		ret.length = strlen((const char *) ret.data);
+	}
+	return ret;
+}
+
+static void lxb_selectors_adapted_set_entry_id_ex(lxb_selectors_entry_t *entry, const lxb_css_selector_t *selector, const xmlNode *node)
+{
+	if (node->doc != NULL && node->doc->dict != NULL) {
+		const xmlChar *interned = xmlDictExists(node->doc->dict, selector->name.data, selector->name.length);
+		if (interned != NULL) {
+			entry->id.name = interned;
+			entry->id.interned = true;
+		} else {
+			entry->id.name = selector->name.data;
+			entry->id.interned = false;
+		}
+	} else {
+		entry->id.name = selector->name.data;
+		entry->id.interned = false;
+	}
+}
+
+static zend_always_inline void lxb_selectors_adapted_set_entry_id(lxb_selectors_entry_t *entry, const lxb_css_selector_t *selector, const xmlNode *node)
+{
+	if (entry->id.name == NULL) {
+		lxb_selectors_adapted_set_entry_id_ex(entry, selector, node);
+	}
+}
+
+static lxb_status_t
+lxb_selectors_state_tree(lxb_selectors_t *selectors, const xmlNode *root,
+						 const lxb_css_selector_list_t *list);
```

Comment:
The formatting of the declarations here seems a bit weird (and also below)

Reply:
Hm, the problem is that github uses tabs that are 8 spaces wide, but my IDE uses tabs that are 4 wide. So it is aligned in my IDE but not on github.

Reply:
Right, well it's a minor issue

Reply:
I can change the broken up lines into spaces and that would fix the issue.

Reply:
Meh, I think I prefer the tabs, and just have a broken GH UI, or need to pass an extra query param to set the tab size